lib/src/app/navigation/app_router_information_parser.dart

```diff
@@ -0,0 +1,52 @@
+import 'dart:convert';
+
+import 'package:allthenews/src/app/navigation/app_path.dart';
+import 'package:flutter/material.dart';
+
+class AppRouteInformationParser extends RouteInformationParser<AppPath> {
+  @override
+  Future<AppPath> parseRouteInformation(RouteInformation routeInformation) async {
+    final uri = Uri.parse(routeInformation.location);
+    return parseRoute(uri);
+  }
+
+  @override
+  RouteInformation restoreRouteInformation(AppPath path) {
+    if (_noArgsPagesConfiguration.containsValue(path)) {
+      final String pagePathLocation = _noArgsPagesConfiguration.entries
+          .firstWhere((entry) => entry.value == path, orElse: () => null)
+          ?.key;
+      return RouteInformation(location: pagePathLocation);
+    } else if (path is WebViewPath) {
+      return RouteInformation(
+        location: '${PagePathLocation.webViewPage}?url=${const HtmlEscape().convert(path.url)}',
```

Comment:
przy przechodzeniu do webview `path.url` jest nullem i leci exception. Z tego co widzę to ta metoda w ogóle nie jest nam potrzebna

Reply:
metoda jako taka potrzebna jest, kiedy ją usunę to na start dostanę kilka exceptionów

Reply:
widzę, że jest już okej, ale też widzę, że w tej  metodzie nic się nie zmieniło, na czym polegał  problem?

Reply:
w setNewRoutePath nie wrzucałem url jako parametr :P